src/renderer/components/ConfigureServer.tsx

```diff
@@ -0,0 +1,355 @@
+    const [url, setUrl] = useState(prevURL || '');
+    const [nameError, setNameError] = useState('');
+    const [urlError, setURLError] = useState('');
+    const [showContent, setShowContent] = useState(false);
+    const [waiting, setWaiting] = useState(false);
+
+    const canSave = name && url && !nameError && !urlError;
+
+    useEffect(() => {
+        setTransition('inFromRight');
+        setShowContent(true);
+    }, []);
+
+    const checkProtocolInURL = (checkURL: string): Promise<string> => {
+        if (urlUtils.startsWithProtocol(checkURL)) {
+            return Promise.resolve(checkURL);
+        }
+
+        return new Promise((resolve) => {
+            let eventCount = 0;
+
+            const handler = (event: {data: {type: string; data: string | Error}}) => {
+                let newURL = checkURL;
+
+                if (event.data.type === PING_DOMAIN_RESPONSE) {
+                    if (event.data.data instanceof Error) {
+                        console.error(`Could not ping url: ${checkURL}`);
+                    } else {
+                        newURL = `${event.data.data}://${checkURL}`;
+                        setUrl(newURL);
+                    }
+
+                    window.removeEventListener('message', handler);
+                    resolve(newURL);
+                } else if (eventCount >= 3) {
+                    window.removeEventListener('message', handler);
+                    resolve(newURL);
+                }
+
+                eventCount++;
+            };
+
+            window.addEventListener('message', handler);
+            window.postMessage({type: PING_DOMAIN, data: checkURL}, window.location.href);
+        });
+    };
+
+    const validateName = () => {
+        const newName = name.trim();
+
+        if (!newName) {
+            return formatMessage({
+                id: 'renderer.components.newTeamModal.error.nameRequired',
+                defaultMessage: 'Name is required.',
+            });
+        }
+
+        if (currentTeams.find(({name: existingName}) => existingName === newName)) {
+            return formatMessage({
+                id: 'renderer.components.newTeamModal.error.serverNameExists',
+                defaultMessage: 'A server with the same name already exists.',
+            });
+        }
+
+        return '';
+    };
+
+    const validateURL = async (fullURL: string) => {
+        if (!fullURL) {
+            return formatMessage({
+                id: 'renderer.components.newTeamModal.error.urlRequired',
+                defaultMessage: 'URL is required.',
+            });
+        }
+
+        if (!urlUtils.startsWithProtocol(fullURL)) {
+            return formatMessage({
+                id: 'renderer.components.newTeamModal.error.urlNeedsHttp',
+                defaultMessage: 'URL should start with http:// or https://.',
+            });
+        }
+
+        if (!urlUtils.isValidURL(fullURL)) {
+            formatMessage({
```

Comment:
I think this is missing a `return` as per my other comment.

Reply:
Good catch, added.